runc/tests/test.bats

```diff
@@ -0,0 +1,25 @@
+@test "Version matches" {
+    # Ensure that `pkg_version` from the plan (embedded in the package
+    # identifier) matches the output from `runc --version`.
+    version="$(echo ${PKGIDENT} | cut -d/ -f3)"
+    result="$(hab pkg exec "${PKGIDENT}" runc --version | head -n1)"
+    [ "$result" = "runc version ${version}" ]
```

Comment:
This file has a mix of `[` and `[[` for comparisons.  I think there was a push at one time to use the more portable `[`, but I'm unsure on the state of this.  If you fancy, perhaps changing these to use `diff` would be the best way to go and to make them consistent.

Reply:
Yup, `diff` would be nicer :+1:

Reply:
Hrmm... actually maybe not. The use of `[[` was so I could have the `=~` operator, allowing me to match a substring. 

As far as portability... if we're using `bats`, that inherently means `bash`, doesn't it?

Reply:
I'm going off of https://github.com/habitat-sh/core-plans/issues/2716 was links to a bats issue with more of the details. Seems less to do with portability and more to do with the behavior of `[[` potentially hiding failures.

For this one in particular I imagine we can use diff though?

Reply:
For this particular one, yes we can certainly use `diff`.

For a bit more color on `[[`, https://github.com/sstephenson/bats/issues/49#issuecomment-48608117 (found following the trail from #2716) indicates that the problematic behavior was fixed in Bash 4; our `core/bats` package, which is what ultimately ends up running these tests, has a dependency on Bash 5. So it seems that the `[` vs `[[` issue isn't really a problem, as far as hiding failing output is concerned.

A bigger problem, reflected in the desire to use `diff`, is getting actionable feedback from failing tests (i.e., something beyond "this assertion failed"). In the past, I've found https://github.com/ztombol/bats-assert to be pretty useful in that regard. It might be useful to see if that could be folded into `core/bats`, or create a `core/bats-assert` and get it and `core/bats` to play nicely.

Reply:
@christophermaier Thanks for digging into that.  I like the idea of getting some assertion library to use across these tests.